src/bp/core/services/middleware/event-engine.ts

```diff
@@ -212,19 +212,23 @@ export class EventEngine {
 
   async replyContentToEvent(
     payload: sdk.Content.All,
-    event: sdk.IO.IncomingEvent,
+    event: sdk.IO.Event,
     options: { incomingEventId?: string; eventType?: string } = {}
   ) {
     payload.metadata = {
       __typing: true,
       ...(payload.metadata ?? {})
     }
 
+    payload = !(event as sdk.IO.IncomingEvent).state
+      ? payload
+      : await this.translatePayload!(payload, event as sdk.IO.IncomingEvent)
+
```

Comment:
Maybe change this to `if (event instanceOf sdk.IO.IncomingEvent)` otherwise it looks bizarre. The OutgoingEvent also has a state variable that isn't present in the typings but would be used here by mistake. The logic should be => if it's and IncomingEvent translate otherwise don't

Reply:
:+1:

Reply:
Ah nevermind then I didn't think of that

Reply:
Is there a direction property or something to differentiate?